artview/components/correlation.py

```diff
@@ -0,0 +1,630 @@
+            for field in self.fieldnames:
+                box.addItem(field)
+
+    def _fieldVerticalAction(self, text):
+        '''Define action for Field Button selection.'''
+        if text == "Field Select":
+            from .field import FieldButtonWindow
+            self.fieldbuttonwindow = FieldButtonWindow(
+                self.Vradar, self.VfieldVertical,
+                name=self.name+"Vertical Field Selection", parent=self.parent)
+        else:
+            self.VfieldVertical.change(str(text))
+
+    def _fieldHorizontalAction(self, text):
+        '''Define action for Field Button selection.'''
+        if text == "Field Select":
+            from .field import FieldButtonWindow
+            self.fieldbuttonwindow = FieldButtonWindow(
+                self.Vradar, self.VfieldHorizontal,
+                name=self.name+"Horizontal Field Selection", parent=self.parent)
+        else:
+            self.VfieldHorizontal.change(str(text))
+
+    def _GateFilterToggleAction(self):
+        '''Define action for GateFilterToggle menu selection.'''
+        if self.gatefilterToggle.isChecked():
+            self.gatefilterToggle.setText("GateFilter On")
+        else:
+            self.gatefilterToggle.setText("GateFilter Off")
+        self._update_plot()
+
+    def _title_input(self):
+        '''Retrieve new plot title.'''
+        val, entry = common.string_dialog_with_reset(
+            self.title, "Plot Title", "Title:", self._get_default_title())
+        if entry is True:
+            self.title = val
+            self._update_plot()
+
+    def _units_input(self):
+        '''Retrieve new plot units.'''
+        val0, entry0 = common.string_dialog_with_reset(
+            self.unitsVertical, "Plot Units", "Vertical Units:",
+            self._get_default_units())
+        val1, entry1 = common.string_dialog_with_reset(
+            self.unitsHorizontal, "Plot Units", "Horizontal Units:",
+            self._get_default_units())
+        if entry0 is True:
+            self.unitsVertical = val0
+        if entry1 is True:
+            self.unitsHorizontal = val1
+        if entry0 is True or entry1 is True:
+            self._update_plot()
+
+    def _add_ImageText(self):
+        '''Add a text box to display.'''
+        from .image_text import ImageTextBox
+        itext = ImageTextBox(self, parent=self.parent)
+        return itext
+
+    def _add_displayBoxUI(self):
+        '''Create the Display Options Button menu.'''
+        parentdir = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                                 os.pardir))
+        config_icon = QtGui.QIcon(os.sep.join(
+            [parentdir, 'icons', "categories-applications-system-icon.png"]))
+        self.dispButton = QtWidgets.QPushButton(config_icon, "", self)
+        self.dispButton.setToolTip("Adjust display properties")
+        self.dispButton.setFocusPolicy(QtCore.Qt.NoFocus)
+        dispmenu = QtWidgets.QMenu(self)
+
```

Comment:
It would be really useful to have controls on the data plot. Including scatter marker type, color, size, and maybe by default remove the edgecolor.

Also, making sure the the regression line bounds to the current axes limits so that the plot resize doesn't shrink the data of interest.